lib/train/transports/cisco_ios_connection.rb

```diff
@@ -23,6 +23,13 @@ def uri
       "ssh://#{@user}@#{@host}:#{@port}"
     end
 
+    def unique_identifier
+      result = run_command_via_connection('show inventory').stdout
+      result.split("\r\n\r\n").each do |section|
+        return section.split('SN: ')[1].strip if section.include?('Chassis')
```

Comment:
Is it possible for Chassie not to be set? If we have multiple boxes like this we will end up having the same uuid of nil on them. If it is possible to be nil we should fall back to a MAC or something.

Reply:
I don't think it is possible for Chassis not to be set. We can't fall back to MAC because modules can change. Also, I don't know how to decided which interface to choose. Example: FastEthernet vs GigabitEthernet.

Reply:
If its always set we should be good. Thanks!